dali/base/dadfs.cpp

```diff
@@ -586,6 +587,56 @@ class CFileSuperOwnerLock : protected CFileLockCompound
     {
         return CFileLockCompound::detach();
     }
+    bool initWithFileLock(const CDfsLogicalFileName &logicalName, IRemoteConnection *conn, unsigned timeout, const char *msg, CFileLock &fcl, unsigned fclmode)
+    {
+        // SuperOwnerLock while holding fcl
+        IRemoteConnection *fclConn = fcl.queryConnection();
+        if (!fclConn)
+            return false; // throw ?
+        CTimeMon tm(timeout);
+        unsigned remaining = timeout;
+        unsigned interval = 5000;
+        loop
+        {
+            try
+            {
+                if (init(logicalName, conn, 0, msg))
+                    return true;
+                else
+                    return false; // throw ?
+            }
+            catch (ISDSException *e)
+            {
+                if (SDSExcpt_LockTimeout != e->errorCode() || tm.timedout(&remaining))
+                    throw;
+                e->Release();
+            }
+            unsigned delta = remaining;
+            if (delta < interval)
+                delta = interval;
+            // release lock
+            {
+                CriticalBlock block(superOwnerLockSect);
+                fclConn->changeMode(RTM_NONE, delta);
+            }
+            tm.timedout(&remaining);
+            unsigned stime = 1000 * (2+getRandom()%15); // 2-15 sec
+            if (stime > remaining)
+                stime = remaining;
+            // let another get excl lock
+            Sleep(stime);
+            tm.timedout(&remaining);
+            delta = remaining;
+            if (delta < interval)
+                delta = interval;
```

Comment:
Not sure why rounding up to 5 secs is good, if user (e.g. via file.lookup) has given a lookup timeout of < 5 seconds or even 0, this forces up to 5 seconds here and again on 2nd changeMode below.
So could in theory wait 10 secs, if user has specified 0.